### ISSUE TITLE
OSX install guide missing important step of sourcing .sh file

### ISSUE THREAD

**Issue by cschneid** [aspnet/Home#499]
brew install dnvm places a shell script that needs to be sourced.

I am using zsh, and the script is entirely silent when run. Only after actually opening up `dnvm.sh` did I realize that it was a bag of functions that needed to be sourced.

Solutions:
1. Update README to note this
2. Have the script output something if it was run, rather than sourced.

**Comment by jrock2004** [comment]
What do you need to add? the path the dvm.sh file itself?

**Comment by jedgrant** [comment]
I'm hoping this is related...
I attempted to follow the OS X installation guidelines. Entering `dnvm update` results in a `-bash: dnvm: command not found` message. Is there a step that I am missing? 

I did...
     `brew tap aspnet/dnx`
     `brew update`
     `brew install dnvm`

 Those commands were successful.

**Comment by miguellira** [comment]
@jedgrant make sure you add `source dnvm.sh` to your `~/.bash_profile`.

Keep in mind, however, that if you've tapped the latest version of Mono (4.0.1) you will run into my issue. See https://github.com/aspnet/Home/issues/498

**Comment by davidcv5** [comment]
After running **brew** and adding **source dnvm.sh** to *~/.zshrc*, I still can't run **dnu restore**.
**dnvm** runs just fine.. but doing **dnvm upgrade** returns the following:

    Determining latest version
    Error: Could not find latest version from feed https://nuget.org/api/v2

Any clue?

**Comment by miguellira** [comment]
@davidcv5 Make sure your NuGet feed is configured to the latest vNext feed, specifically `https://www.myget.org/F/aspnetvnext/api/v2`. On Mac OS X this is located in: `~/.config/NuGet/NuGet.Config `

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="AspNetVNext" value="https://www.myget.org/F/aspnetvnext/api/v2" />
    <add key="NuGet.org" value="https://nuget.org/api/v2/" />
  </packageSources>
</configuration>
```

**Comment by jrock2004** [comment]
dnvm upgrade -u

**Comment by davidcv5** [comment]
wow... that was fast!!! Thanks a lot @miguellira and @jrock2004 

I tired @jrock2004  solution which was shorter and it worked prefect!!

Thanks a lot!

**Comment by miguellira** [comment]
np @davidcv5 Do you mind executing `mono --version` and letting me know what version of Mono you are running against?

**Comment by davidcv5** [comment]
@miguellira .. I'm running 4.0.1

**Comment by jedgrant** [comment]
@miguellira please pardon my lack of experience, how do I add that? `nano .bash_profile` I assume? But what should the string look like? Currently it is `export PATH="~/npm-global/bin:$PATH"` does it just need to change like... `export PATH="~/npm-global/bin:source dnvm.sh:$PATH"`?

**Comment by davidcv5** [comment]
@jedgrant you can just add it as a new line.
so you're new file will look like this...

    export PATH="~/npm-global/bin:$PATH"
    source dnvm.sh

**Comment by jedgrant** [comment]
Fantastic, thank you!

**Comment by TomSchillemans** [comment]
I've added the `source dmvm.sh` to my `~/.zshrc` and run @jrock2004's line and it all works perfectly now!! So basically this needs to be mentioned in the README and it all is fine and dandy.

**Comment by chemitaxis** [comment]
Hi! How can I add it to my zshrc?? Thanks!!

**Comment by chemitaxis** [comment]
I have done all in this post, but I still have the same problem... using dnvm upgrade and dnvm upgrade -u and dnvm upgrade -f (force).

Error: Could not find latest version from feed https://nuget.org/api/v2

But have the same problem.. maybe something wrong adding to my ~/.zshrc, what is the correct way please?

Thanks!!

**Comment by davidcv5** [comment]
@chemitaxis If you get that message that means the dnvm command was found, so your .zshrc file should be fine.. 
Using **dnvm upgrade -u** worked perfect for me, but if yours is still trying to pull from https://nuget.org/api/v2, I would suggest setting your NuGet config file as previously suggested by **miguellira**.

**Comment by chemitaxis** [comment]
HI @davidcv5, I have done all, changed the *~/.config/NuGet/NuGet.Config* that just had this

```<?xml version="1.0" encoding="utf-8"?>
<configuration>
</configuration>```

Adding that, but no way...

```<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="AspNetVNext" value="https://www.myget.org/F/aspnetvnext/api/v2" />
    <add key="NuGet.org" value="https://nuget.org/api/v2/" />
  </packageSources>
</configuration>```

I don't know what I did bad...

**Comment by misterzik** [comment]
would it be okay, if i only add it to one of the following three files? or does it have to be in all of them?
~/.bash_profile, ~/.bashrc or ~/.zshrc

source dnvm.sh

Best Regards

**Comment by miguellira** [comment]
@misterzik it all depends on the shell you're using. This page sums up your options: [.bash_profile vs .bashrc](http://www.joshstaiger.org/archives/2005/07/bash_profile_vs.html)

Basically, if you use **Terminal** on Mac OS X you may want to add this to your `~/.bash_profile`
```
if [ -f ~/.bashrc ]; then
   source ~/.bashrc
fi
```
and this in your `~/.bashrc`
```
source dnvm.sh
```

**Comment by misterzik** [comment]
@miguellira Thanks Miguel, I actually did something bit different and it seems to be working for now, but incase it stops working, I'll make sure to redirect to this page.

I used this on my ~/.bashrc ;

    echo "source dnvm.sh" >> ~/.bashrc

then restarted it my terminal and was working just fine, but thanks anyways for taking your time to respond. Cheers

**Comment by joeblew99** [comment]
Hey, i have same issue and so wrote a test .sh script-

tool-init-env.1.sh
````
#!/bin/bash

source $PWD/tool-init-env.2.sh

source $HOME/workspace/go/src/bitbucket.org/apparently_different/backend/tool-init-env.2.sh

echo "YOUR PATH:"
echo $PATH
```

tool-init-env.2.sh
````
#!/bin/bash

export PATH=$PATH:/User/blahblahblahblahblah
```

Now check what happens when its run:

````
gedw99-MacBook-Pro:backend apple$ ./tool-init-env.1.sh 
YOUR PATH:
/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin:/usr/local/go/bin:/Users/apple/workspace/go/bin:/User/blahblahblahblahblah:/User/blahblahblahblahblah
gedw99-MacBook-Pro:backend apple$ echo $PATH
/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin:/usr/local/go/bin:/Users/apple/workspace/go/bin
gedw99-MacBook-Pro:backend apple$ 

```

So whats happening ?

when script 1 does it thing and then echos the $PATH, it has the new path

But straight after, when you do a echo $PATH in terminal the new path is gone.


On Linux it behaves the same too:
````
vagrant@vagrant-ubuntu-trusty-64:~$ ./1.sh
YOUR PATH:
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/User/blahblahblahblahblah
vagrant@vagrant-ubuntu-trusty-64:~$ echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
vagrant@vagrant-ubuntu-trusty-64:~$ 

```

**Comment by miguellira** [comment]
@joeblew99 Try using `source` to execute the script commands in the current shell.

Instead of:
```bash
gedw99-MacBook-Pro:backend apple$ ./tool-init-env.1.sh 
```
try
```bash
gedw99-MacBook-Pro:backend apple$ source tool-init-env.1.sh 
```
But really, now that https://get.asp.net/ is live, getting the appropriate package for your given OS is probably the best way to get started. Just be prepared to start all over again once `donet cli` is official: http://www.hanselman.com/blog/ExploringTheNewNETDotnetCommandLineInterfaceCLI.aspx

**Comment by joeblew99** [comment]
Hey @mig.

Yes I am aware that from the terminal directly it does work. But I actually want to be able to call a .sh that has the ability inside the script to set the path mappings for the duration if that bash session.

It much easier for the team to use is the reason. And I just want to know how its done :)

**Comment by aspnet-hello** [comment]
This issue is being closed because it has not been updated in 3 months.

We apologize if this causes any inconvenience. We ask that if you are still encountering this issue, please log a new issue with updated information and we will investigate.
